lib/src/component_declaration/component_base.dart

```diff
@@ -446,6 +702,46 @@ abstract class UiStatefulComponent<TProps extends UiProps, TState extends UiStat
   // ----------------------------------------------------------------------
 }
 
+abstract class UiStatefulComponent2<TProps extends UiProps, TState extends UiState> extends UiComponent2<TProps> {
```

Comment:
Does this need to implement `UiStatefulComponent`?

Also - this deserves a doc comment since it is public via the `component_base.dart` entrypoint.